rules/ruleset.go

```diff
@@ -41,10 +41,30 @@ var (
 	errNilRuleSetSchema = errors.New("nil rule set schema")
 )
 
+// MatchMode determines how match is performed.
+type MatchMode int
+
+// List of supported match modes.
+const (
+	// When performing matches in MappingPoliciesOnly mode, the matcher only finds the applicable
+	// mapping policies for the given id, without attempting to determine the rollup policies.
+	MappingPoliciesOnly MatchMode = iota
+
+	// When performing matches in Full mode, the matcher find the applicable mapping
+	// policies as well as the rollup policies for the given id.
+	Full
+)
+
+// MatchOptions provide a set of options for id matching.
+type MatchOptions struct {
+	Mode                MatchMode
+	UseDefaultIfNoMatch bool
+}
+
 // Matcher matches metrics against rules to determine applicable policies.
 type Matcher interface {
-	// MatchAll returns all applicable policies given a metric id between [from, to).
-	MatchAll(id []byte, from time.Time, to time.Time) MatchResult
+	// MatchAll returns the applicable policies for a metric id between [from, to).
+	MatchAll(id []byte, isRollupID bool, from, to time.Time, opts MatchOptions) MatchResult
```

Comment:
`MatchAll` refers to the fact that it matches policies at different times within the range. I could change it to `MatchRange` if you feel strongly?